fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/SearchFilter.java

```diff
@@ -42,6 +42,22 @@
         QueryType(String type) {
             this.type = type;
         }
+
+        /**
+         * Get the QueryType enum from it's search type.
+         *
+         * @param type  The type of query (i.e. "insensitive_contains").
+         *
+         * @return the enum QueryType.
+         */
+        public static QueryType fromType(String type) {
```

Comment:
@QubitPi Are you fine with me reverting to the [IllegalArgumentException](https://github.com/yahoo/fili/pull/396/commits/0ced60b2512fc2eb621603e92d84bac4ca110395#r123132356)/returning optional?

For context, this was made because `SearchFilter` doesn't have a reference to its `QueryType` and I have to process a `SearchFilter` differently for each case (and if not found, throw an exception).

Additionally, if `SearchFilter` must be one of those `QueryType` then it should probably be verified in the [constructor](https://github.com/yahoo/fili/blob/369787a86097e19c672072c8fe4e5b76e4fc012a/fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/SearchFilter.java#L67) that accepts a `String` queryType.